datafusion/core/tests/sql/parquet.rs

```diff
@@ -165,6 +165,39 @@ async fn fixed_size_binary_columns() {
     }
 }
 
+#[tokio::test]
+async fn window_fn_timestamp_tz() {
+    let ctx = SessionContext::new();
+    ctx.register_parquet(
+        "t0",
+        "tests/parquet/data/timestamp_with_tz.parquet",
+        ParquetReadOptions::default(),
+    )
+    .await
+    .unwrap();
+
+    let sql = "SELECT count, LAG(timestamp, 1) OVER (ORDER BY timestamp) FROM t0";
+    let dataframe = ctx.sql(sql).await.unwrap();
+    let results = dataframe.collect().await.unwrap();
+
+    let mut num_rows = 0;
+    for batch in results {
+        num_rows += batch.num_rows();
+        assert_eq!(2, batch.num_columns());
+
+        let ty = batch.column(0).data_type().clone();
+        assert_eq!(DataType::Int64, ty);
+
+        let ty = batch.column(1).data_type().clone();
+        assert_eq!(
+            DataType::Timestamp(TimeUnit::Millisecond, Some("UTC".to_owned())),
+            ty
+        );
+    }
+
+    assert_eq!(131072, num_rows);
```

Comment:
👍 


In the future you could also use sqllogictest to write this kind of test. Something like https://github.com/apache/arrow-datafusion/blob/eb9541380964a5fd5aa92302a82ec189c3be792e/datafusion/core/tests/sqllogictests/test_files/window.slt#L18-L36